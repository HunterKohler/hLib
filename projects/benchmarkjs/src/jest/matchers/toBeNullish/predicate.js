export default recieved => recieved === null || recieved === undefined;
