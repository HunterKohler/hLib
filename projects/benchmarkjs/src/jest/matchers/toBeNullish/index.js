import { matcherHint, printReceived, printExpected } from "jest-matcher-utils";
import predicate from "./predicate";

const passMessage = recieved => () =>
    matcherHint(".not.toBeNullish", "recieved", "") 
    + "\n\n"
    + `Expected value to not be nullish, recieved:`
    + `    ${printReceived(recieved)}`;

const failMessage = recieved => () =>
    matcherHint(".toBeNullish", "recieved", "")
    + "\n\n"
    + "Expected value to be nullish, recieved:"
    + `    ${printReceived(recieved)}`

export default function toBeNullish(recieved) {
    const pass = predicate(recieved);
    return { pass, message: pass
        ? passMessage(recieved)
        : failMessage(recieved)
    };
}
