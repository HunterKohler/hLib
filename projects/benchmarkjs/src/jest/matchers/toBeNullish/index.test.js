import toBeNullish from "./";

expect.extend({ toBeNullish });

describe(".toBeNullish", () => {
    test.each([
        [undefined], [null]
    ])("pass when given %s", value => {
        expect(value).toBeNullish();
    });

    test.each([
        [false], [0], [1], [[]], [{}], [NaN], [""]
    ])("fail when given %s", value => {
        expect(() => expect(value).toBeNullish()).toThrowErrorMatchingSnapshot();
    });
});

describe(".not.toBeNullish", () => {
    test.each([
        [undefined], [null]
    ])("fail when given %s", value => {
        expect(() => expect(value).not.toBeNullish()).toThrowErrorMatchingSnapshot();
    });

    test.each([
        [false], [0], [1], [[]], [{}], [NaN], [""]
    ])("pass when given %s", value => {
        expect(value).not.toBeNullish()
    });
});
