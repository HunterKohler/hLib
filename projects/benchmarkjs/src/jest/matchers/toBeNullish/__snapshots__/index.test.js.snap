// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toBeNullish fail when given null 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).not.toBeNullish()[22m

Expected value to not be nullish, recieved:    [31mnull[39m"
`;

exports[`.not.toBeNullish fail when given undefined 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).not.toBeNullish()[22m

Expected value to not be nullish, recieved:    [31mundefined[39m"
`;

exports[`.toBeNullish fail when given  1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31m\\"\\"[39m"
`;

exports[`.toBeNullish fail when given [] 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31m[][39m"
`;

exports[`.toBeNullish fail when given {} 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31m{}[39m"
`;

exports[`.toBeNullish fail when given 0 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31m0[39m"
`;

exports[`.toBeNullish fail when given 1 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31m1[39m"
`;

exports[`.toBeNullish fail when given NaN 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31mNaN[39m"
`;

exports[`.toBeNullish fail when given false 1`] = `
"[2mexpect([22m[31mrecieved[39m[2m).toBeNullish()[22m

Expected value to be nullish, recieved:    [31mfalse[39m"
`;
